flink-runtime/src/main/java/org/apache/flink/runtime/failure/FailureEnricherUtils.java

```
@@ -0,0 +1,227 @@
+        Set<String> includedEnrichers = getIncludedFailureEnrichers(configuration);
+        //  When empty, NO enrichers will be started.
+        if (includedEnrichers.isEmpty()) {
+            return Collections.emptySet();
+        }
+        final Iterator<FailureEnricherFactory> factoryIterator =
+                pluginManager.load(FailureEnricherFactory.class);
+        final Set<FailureEnricher> failureEnrichers = new HashSet<>();
+        while (factoryIterator.hasNext()) {
+            try {
+                final FailureEnricherFactory failureEnricherFactory = factoryIterator.next();
+                final FailureEnricher failureEnricher =
+                        failureEnricherFactory.createFailureEnricher(configuration);
+                if (includedEnrichers.contains(failureEnricher.getClass().getName())) {
+                    failureEnrichers.add(failureEnricher);
+                    LOG.debug(
+                            "Found failure enricher {} at {} ",
+                            failureEnricherFactory.getClass().getName(),
+                            new File(
+                                            failureEnricher
+                                                    .getClass()
+                                                    .getProtectionDomain()
+                                                    .getCodeSource()
+                                                    .getLocation()
+                                                    .toURI())
+                                    .getCanonicalPath());
+                } else {
+                    LOG.info(
+                            "Excluding failure enricher {}, not configured in enricher list ({}).",
+                            failureEnricherFactory.getClass().getName(),
+                            includedEnrichers);
+                }
+            } catch (Exception e) {
+                LOG.warn("Error while loading failure enricher factory.", e);
```

I'd personally throw an exception here because this violates the cluster setup that the user has asked for. Also, since this happens during JM startup, it should be fine to fail fast.

---

I agree, but lets leave it as is for now and add the exception when we decide where we are call the factory (JM/Dispatcher/ClusterEntryPoint)